### ISSUE TITLE
Accessibility in Introduction section of API Reference

### ISSUE THREAD

**Issue by kayla-beckham** [scalar/scalar#3422]
### What happens?

![Screen Shot 2024-10-09 at 4 19 55 PM](https://github.com/user-attachments/assets/845a5acd-eaf2-4047-9899-3df001cb5dc3)

1. There is no visual indication of focus when the keyboard navigates over Base Url and authentication fields in the introduction card of an API Reference. The typing cursor appears in Authentication fields, but there should be a high contrast focus box that shows when we navigate to all of these active elements.
2. In the Introduction card, the labels of the Base Url, authentication type, and Client Libraries sections should be announced by a screen reader when a user interacts with the fields.
3. In authentication section in the introduction card, screen readers do not announce the current state of the ‘show password’ button (show or hide).
4. In the Client Libraries selection tab menu, the focus goes only to the More combobox, but the focus is not visually indicated in a high-contrast way. I can navigate the library drop down combobox with the keyboard, but focus does not go through the other tabs displayed horizontally. 
    1. When I move the keyboard focus to the More button in Client Libraries, the screen reader announces the name of the library that is currently selected. This is not a meaningful announcement. The screen reader needs to give more context about the purpose of this menu.
    2. The icons besides languages in the horizontal tabs of the Client Libraries menu are missing the following attributes: focusable="false”, role="presentation”, aria-hidden="true"
    3. The tab elements are missing the following attributes: role="tablist" for tab container, role="tab" for tab, role="tabpanel" for page under the tab, aria-control - for each tab referring to associated tabpanel, aria-select - true for the current page, and false for others
5. 'Download OpenAPI Document' button should have role="link" as the main function of it is to download a document. Should have attribute role="link", and screenreader should announce the 'Download OpenAPI Document' as a link.
6. There is no ‘main’ landmark on the page. 

### What did you expect to happen?

All active elements in the introduction card should be keyboard accessible.
When the keyboard focus is on an element, there should be a visual indicator of the focus (usually a box highlighting the element). 
The screen reader should be able to make an announcement of each element that includes context for the visual label.  For example,  the Client Libraries menu options should be able to say 'Client Libraries Curl'. 
Buttons for showing or hiding elements should have screen reader announcements for the status. 
The Download OpenAPI Document button should have role=link. 
I think the 'main' landmark should wrap the introduction section and all the sections for tags/endpoints, but you may decide to have it only around the tag sections. There needs to be one somewhere on the page.

**Comment by hwkr** [comment]
Hey @kayla-beckham! I've made some updates to Anthony's PR based on your feedback. A few things:

* I didn't add `focusable: false` to the icons since as far as I can tell the only browsers that try to focus `<svg>`'s are IE and Edge v13 and earlier (which we don't support)
* For the client libraries I disabled the featured libraries (the ones in the tabs) via keybard and just updated the `<select>` to be screen reader compatible. The `<select>` contains all the options in the featured tabs and is simpler in my opinion that having a complex multi input widget for screen readers. We have a combobox component that I wouldn't mind putting in there in the future but I didn't want to add that to this PR.
* I didn't add a `main` landmark to the core app because it is often embedded in other pages (which may already define a `main` landmark) and nesting `main` landmarks is against ARIA spec (see [comment](https://github.com/scalar/scalar/pull/3435#discussion_r1796056028)).
